migration/planning-migration-3-to-4.adoc

```diff
@@ -0,0 +1,194 @@
+
+[discrete]
+==== Infrastructure options
+
+In {product-title} 3.11, you installed your cluster on infrastructure that you prepared and maintained. In addition to providing your own infrastructure, {product-title} 4 offers an option to deploy a cluster on infrastructure that the {product-title} installation program provisions and the cluster maintains.
+
+For more information, see xref:../architecture/architecture-installation.adoc#installation-overview_architecture-installation[OpenShift Container Platform installation overview].
+
+[discrete]
+==== Upgrading your cluster
+
+In {product-title} 3.11, you upgraded your cluster by running Ansible playbooks. In {product-title} 4.2, The cluster manages its own updates, including updates to {op-system-first} on cluster nodes. You can easily upgrade your cluster by using the web console or by using the `oc adm upgrade` command from the OpenShift CLI and the Operators will automatically upgrade themselves. If your {product-title} 4.2 cluster has Red Hat Enterprise Linux workers, then you will still need to run an Ansible playbook to upgrade those workers.
+
+For more information, see xref:../updating/updating-cluster-between-minor.adoc#updating-cluster-between-minor[Updating clusters].
+
+[id="migration-considerations"]
+== Migration considerations
+
+Review the changes and other considerations that might affect your migration from {product-title} 3.11 to {product-title} 4.
+
+[id="migration-preparing-storage"]
+=== Storage considerations
+
+Review the storage changes to consider when migrating from {product-title} 3.11 to {product-title} 4.2.
+
+[discrete]
+==== Local volume persistent storage
+
+Local storage is only supported by using the Local Storage Operator in {product-title} 4.2. It is not supported to use the local provisioner method from {product-title} 3.11.
+
+For more information, see xref:../storage/persistent-storage/persistent-storage-local.adoc#persistent-storage-using-local-volume[Persistent storage using local volumes].
+
+[discrete]
+==== FlexVolume persistent storage
+
+The FlexVolume plug-in location has been changed to a new location from {product-title} 3.11.  The new location in {product-title} 4.2 is `/etc/kubernetes/kubelet-plugins/volume/exec`. Attachable FlexVolume plug-ins are no longer supported.
+
+// TODO: No FlexVolume docs in 4.2 to link to right now.
+
+[discrete]
+==== Container Storage Interface (CSI) persistent storage
+
+Persistent storage using the Container Storage Interface (CSI) was technology preview in {product-title} 3.11. CSI version 1.1.0 is fully supported in {product-title} 4.2, but does not ship with any CSI drivers. You must install your own driver.
```

Comment:
was in link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] in